pkg/cvo/cvo_scenarios_test.go

```diff
@@ -1566,6 +1566,233 @@ func TestCVO_UpgradeUnverifiedPayload(t *testing.T) {
 	})
 }
 
+func TestCVO_UpgradeFailedPayloadLoadWithCapsChanges(t *testing.T) {
+	o, cvs, client, _, shutdownFn := setupCVOTest("testdata/payloadtest-2")
+
+	// Setup: a successful sync from a previous run, and the operator at the same image as before
+	//
+	o.release.Image = "image/image:0"
+	o.release.Version = "1.0.0-abc"
+	desired := configv1.Release{Version: "1.0.1-abc", Image: "image/image:1"}
+	uid, _ := uuid.NewRandom()
+	clusterUID := configv1.ClusterID(uid.String())
+	cvs["version"] = &configv1.ClusterVersion{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:            "version",
+			ResourceVersion: "1",
+			Generation:      1,
+		},
+		Spec: configv1.ClusterVersionSpec{
+			ClusterID:     clusterUID,
+			Channel:       "fast",
+			DesiredUpdate: &configv1.Update{Version: desired.Version, Image: desired.Image},
+			Capabilities: &configv1.ClusterVersionCapabilitiesSpec{
+				BaselineCapabilitySet:         configv1.ClusterVersionCapabilitySetNone,
+				AdditionalEnabledCapabilities: []configv1.ClusterVersionCapability{"baremetal"},
```

Comment:
nit: we have `configv1.ClusterVersionCapabilityBaremetal`, which is a lot longer than `"baremetal"`, but gets us compile-time typo protection.  And maybe helps folks grepping for consumers?  Doesn't seem like a big win, so update or leave your current string as it stands, as you see fit.